android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java

```diff
@@ -1,67 +1,215 @@
+      response.putString(Utils.HAS_WALLET_INSTALLED, "");
+      response.putInt(Utils.RESPONSE_CODE, ResponseCode.OK.getValue());
+      return response;
+    }
   }
 
   @Override public Bundle getPurchases(int apiVersion, String packageName, String type,
       String continuationToken) {
-
     Bundle bundleResponse = new Bundle();
+    if (WalletUtils.hasWalletInstalled()) {
+      try {
+        if (!wasServiceBinded) {
+          createRepository();
+          synchronized (lockThread) {
+            lockThread.wait();
+          }
+        }
+        return serviceAppcoinsBilling.getPurchases(apiVersion, packageName, type, null);
+      } catch (RemoteException e) {
+        e.printStackTrace();
+        bundleResponse.putInt(Utils.RESPONSE_CODE, ResponseCode.SERVICE_UNAVAILABLE.getValue());
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+        bundleResponse.putInt(Utils.RESPONSE_CODE, ResponseCode.SERVICE_UNAVAILABLE.getValue());
+      }
+    } else {
 
-    bundleResponse.putInt(Utils.RESPONSE_CODE, ResponseCode.OK.getValue());
-    bundleResponse.putStringArrayList(Utils.RESPONSE_INAPP_PURCHASE_DATA_LIST,
-        new ArrayList<String>());
-    bundleResponse.putStringArrayList(Utils.RESPONSE_INAPP_SIGNATURE_LIST, new ArrayList<String>());
-    bundleResponse.putStringArrayList(Utils.RESPONSE_INAPP_PURCHASE_ID_LIST,
-        new ArrayList<String>());
+      bundleResponse.putInt(Utils.RESPONSE_CODE, ResponseCode.OK.getValue());
+      bundleResponse.putStringArrayList(Utils.RESPONSE_INAPP_PURCHASE_DATA_LIST,
+          new ArrayList<String>());
+      bundleResponse.putStringArrayList(Utils.RESPONSE_INAPP_SIGNATURE_LIST,
+          new ArrayList<String>());
+      bundleResponse.putStringArrayList(Utils.RESPONSE_INAPP_PURCHASE_ID_LIST,
+          new ArrayList<String>());
+    }
 
     return bundleResponse;
   }
 
   @Override public int consumePurchase(int apiVersion, String packageName, String purchaseToken) {
-    return ResponseCode.OK.getValue();
+
+    if (WalletUtils.hasWalletInstalled()) {
+      try {
+        if (!wasServiceBinded) {
+          createRepository();
+          synchronized (lockThread) {
+            lockThread.wait();
+          }
+        }
+        return serviceAppcoinsBilling.consumePurchase(apiVersion, packageName, purchaseToken);
+      } catch (RemoteException e) {
+        e.printStackTrace();
+        return ResponseCode.SERVICE_UNAVAILABLE.getValue();
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+        return ResponseCode.SERVICE_UNAVAILABLE.getValue();
+      }
+    } else {
+      return ResponseCode.OK.getValue();
+    }
   }
 
   @Override public IBinder asBinder() {
     return null;
   }
 
+  private void createRepository() {
+    final AppCoinsAndroidBillingRepository repository =
+        new AppCoinsAndroidBillingRepository(Utils.API_VERSION, WalletUtils.walletPackageName);
+
+    RepositoryServiceConnection connection =
```

Comment:
if we are including RepositoryServiceConnection class  here, it can't implement RepositoryConnection because there is no reason to add this interface to migrator's generated code